modal/image.py

```diff
@@ -156,11 +156,29 @@ def is_str_list(x):
     return ret
 
 
+def _validate_packages(packages: List[str]) -> bool:
+    """Validates that a list of packages does not contain any command-line options."""
+    return not any(pkg.startswith("-") for pkg in packages)
+
+
+def _warn_invalid_packages(old_command: str) -> None:
+    deprecation_warning(
+        (2024, 7, 3),
+        "Passing flags to `pip` via the `packages` argument of `pip_install` is deprecated."
+        + " Please pass flags via the `extra_options` argument instead."
+        + "\nNote that this will require a rebuild of the image."
+        + " To avoid rebuilding, use the exact same `pip install` command via `run_commands`, i.e.:"
```

Comment:
```suggestion
        + " To avoid rebuilding, you can pass the following to `run_commands` instead:"
```
suggesting a small rewording to make it more clear that this is an optional remediation